osu.Game/Online/Spectator/SpectatingUserState.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Online.Spectator
+{
+    public enum SpectatingUserState
```

Comment:
This name is confusing me - the enum is talking about "spectating user" state, and the xmldocs are talking about "spectated user" state. In my mind those are two different users.

Reply:
Yeah I think the name `SpectatedUserState`/`SpectatedPlayerState` suits better.

Reply:
I've renamed `SpectatingUserState` -> `SpectatedUserState`. I think there is validity to renaming it to `GameplayCompletionState` and replacing the bools in `GameplayState` with it, however that is a refactoring which touches many files and I think is better to be kept for a separate PR if done at all.